x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/comparison_side/versions_picker/constants.ts

```diff
@@ -24,35 +29,120 @@ export enum SelectedVersions {
   TargetFinal = 'target_final',
 }
 
-export const CURRENT_OPTIONS: EuiSelectOption[] = [
-  {
-    value: SelectedVersions.CurrentFinal,
-    text: i18n.VERSION1_VS_VERSION2(i18n.CURRENT_VERSION, i18n.FINAL_VERSION),
-  },
-  {
-    value: SelectedVersions.CurrentTarget,
-    text: i18n.VERSION1_VS_VERSION2(i18n.CURRENT_VERSION, i18n.TARGET_VERSION),
-  },
-];
-
-export const TARGET_OPTIONS: EuiSelectOption[] = [
-  {
-    value: SelectedVersions.TargetFinal,
-    text: i18n.VERSION1_VS_VERSION2(i18n.TARGET_VERSION, i18n.FINAL_VERSION),
-  },
-];
-
-export const BASE_OPTIONS: EuiSelectOption[] = [
-  {
-    value: SelectedVersions.BaseFinal,
-    text: i18n.VERSION1_VS_VERSION2(i18n.BASE_VERSION, i18n.FINAL_VERSION),
-  },
-  {
-    value: SelectedVersions.BaseTarget,
-    text: i18n.VERSION1_VS_VERSION2(i18n.BASE_VERSION, i18n.TARGET_VERSION),
-  },
-  {
-    value: SelectedVersions.BaseCurrent,
-    text: i18n.VERSION1_VS_VERSION2(i18n.BASE_VERSION, i18n.CURRENT_VERSION),
-  },
-];
+export const getOptionsForDiffOutcome = (
+  fieldDiff: ThreeWayDiff<unknown>,
+  resolvedValue: unknown
+): Array<{ value: SelectedVersions; text: string; title: string }> => {
+  switch (fieldDiff.diff_outcome) {
+    case ThreeWayDiffOutcome.StockValueCanUpdate: {
+      const hasUserChangedResolvedValue = !isEqual(fieldDiff.merged_version, resolvedValue);
+
+      const options = [
+        {
+          value: SelectedVersions.CurrentTarget,
```

Comment:
It's quite confusing to see `current vs target` while the rest options use `base vs target` to Elastic updates. Have you considered changing the implementation to improve readability?

It looks much more straightforward to stick to concrete selector values `MyChanges`, `UpdateFromElastic` and `Merged`. Decision on what exact versions are used for comparison could implemented in a separate function with the body pretty similar to `getOptionsForDiffOutcome`.

There are the following comparison options when base version exists

- `MyChanges` shows `base vs final`
- `UpdateFromElastic` shows `base vs target`
- `Merged` shows `base_final` (`Merged` replaces `MyChanges` for solvable conflicts)

In case of missing base version
- `UpdateFromElastic` shows `current vs target`
- `MyChanges` shows `current vs final`